tools/build_defs/framework.bzl

```diff
@@ -404,23 +429,29 @@ def _define_inputs(attrs):
     linking_infos_all = []
 
     bazel_headers = []
+    bazel_system_includes = []
     bazel_libs = []
 
     # This framework function-built libraries: copy result directories under
     # $EXT_BUILD_DEPS/lib-name
     ext_build_dirs = []
+    ext_build_dirs_set = {}
 
     for dep in attrs.deps:
-        provider = dep[OutputGroupInfo]
-        ext_built = provider and hasattr(provider, "gen_dir")
+        provider = dep[ExternallyBuiltTransitive]
 
         linking_infos_all += [dep[CcLinkingInfo]]
         compilation_infos_all += [dep[CcCompilationInfo]]
 
-        if ext_built:
-            ext_build_dirs += provider.gen_dir.to_list()
+        if provider:
```

Comment:
Is this loop not flattening the depset for every library?

Reply:
provider == null if the library was built with Bazel
Yep by the way you are right, we should consider iterating the depset in the place where the script text is created.
Could we do it as a separate PR?